### PR TITLE
Add support for dashboard descriptions

### PR DESCRIPTION
**Pull request by MisterSquishy**
adds support for dashboard descriptions, via the `dashboard_description` field. also updates accompanying documentation and tests

### CODE REVIEW

File: docs/index.md [hunk 1]
```diff
@@ -22,12 +22,9 @@ provider "lightstep" {
...
-### Required
-
-- `organization` (String) The name of the Lightstep organization
-
 ### Optional
 
 - `api_key` (String) The API Key for a Lightstep organization.
 - `api_key_env_var` (String) Environment variable for Lightstep api key.
 - `environment` (String) The name of the Lightstep environment, must be one of: staging, meta, public.
+- `organization` (String) The name of the Lightstep organization
```
Comment (MisterSquishy) [inline docs/index.md:30]: looks like this got moved down here to reflect the fact that this required arg has a default (falls back to the environment key) https://github.com/lightstep/terraform-provider-lightstep/blob/c50ea86e0ea1f9019758ed6d732dd75e90b9a8d9/lightstep/provider.go#L24

i dont really have an opinion, the plugin did this. im happy to move it back

File: .go-version [hunk 2]
```diff
@@ -1 +1 @@
-1.70.7
```
Comment (bcronin) [inline .go-version:1]: Are you sure about this? :)

**Review by bcronin — APPROVED** [review]
Looks good!

I assume we need to wait until the public API support for this field is fully deployed before this can be merged (or, more specifically, before this version is tagged as a new release).

File: .go-version (hunk 2 above)
Comment (MisterSquishy) [reply .go-version:1]: wait im so paranoid from screwing this up last time... is it wrong??

**Comment by MisterSquishy** [comment]
> Looks good!
> 
> I assume we need to wait until the public API support for this field is fully deployed before this can be merged (or, more specifically, before this version is tagged as a new release).

public api support is fully deployed!

File: .go-version (hunk 2 above)
Comment (bcronin) [reply .go-version:1]: Sorry! Was just joking given our conversation about the version numbers!  This is :+1:!!!

Comment (MisterSquishy) [reply .go-version:1]: hahaha i thought so but i didnt want to ignore it and screw up again 😅

**Pull request merged**